tests/MongoDB.Driver.Tests/Specifications/auth/AuthTestRunner.cs

```diff
@@ -109,7 +113,19 @@ private void AssertValid(MongoCredential mongoCredential, BsonDocument definitio
                 }
                 else
                 {
-                    // Other authenticators do not contain mechanism properties
+                    if (expectedCredential.TryGetValue("mechanism_properties", out var expectedMechanismProperties))
```

Comment:
I believe these two code paths can be merged into one. But I will rely on your and other reviewers' opinion here

Reply:
There does seem to be some duplication.

But it's also slightly different. One path verifies the mechanism properties of the mechanism itself (but only for a GssapAuthenticator) while the other path verifies the mechanism properties of the credential itself.

Reply:
The difference is that for `GSSAPI` mechanism the code sets `SERVICE_NAME` and `CANONICALIZE_HOST_NAME` properties only after creating an `Authenticator` here: [GssapiAuthenticator.CreateMechanism()](https://github.com/mongodb/mongo-csharp-driver/blob/0d695d517fade7d70c207b1afbf10b5306a7d210/src/MongoDB.Driver.Core/Core/Authentication/GssapiAuthenticator.cs#L146). And these two properties are not taken directly from connection string but are expected by specification which is seen in the test: [should accept valid authSource (GSSAPI)](https://github.com/mongodb/specifications/blob/11970bc05b87ae58d8783d66b2d1eb29efdec04b/source/auth/tests/connection-string.json#L68).
Other mechanisms tests do not expect any properties besides those contained in connection string.